### DISCUSSION TITLE
How do I extend more epochs once training's done?

### DISCUSSION THREAD

**Discussion by qjygsantos** [ultralytics/ultralytics#16092] [General]
I just finished training a YOLOv5 model. It was set at 0.001 for both lr0 and lr1, 0.9 momentum, batch size of 8, and 300 epochs (other config are default). I trained it from scratch. I did the training not continuously since Im limited with Google Colab's free plan. I used --resume parameter.

It did work and continue training where it left off as long as I'm calling the directory and last.pt file correctly. 

My question is, now that 300 epochs is done. I noticed that before it ended the mAP is still increasing and now I want to extend it for 400 epochs. 

Is this possible? or I need to train a new model again from the start?? Because I want the same configurations and all. Like I just want to continue from the model or training's state before it actually ended. Like epoch 301 and so on.

**Comment by glenn-jocher** [comment]
You can continue training from where you left off by using the `--resume` parameter with your `last.pt` file. Just set the `epochs` parameter to 400 in your command, and it will continue from epoch 301.

**Reply by qjygsantos** [reply]
Hi sorry if my post sounds confusing but what I meant is that I originally set the epochs to 300 and now training is done.... the mAP is still increasing before it ended and now I want to add 100 more epochs.

Is it possible to do that or I need to start training all over again from epoch 0 and set the number of epochs to 400?

**Reply by glenn-jocher** [reply]
@qjygsantos you can continue training by using the `--resume` parameter with your `last.pt` file and setting `epochs` to 400. This will pick up from where it left off.
